protocol/xmrtaker/instance.go

```diff
@@ -271,23 +271,12 @@ func (inst *Instance) maybeCancelNewSwap(txHash ethcommon.Hash) (bool, error) {
 
 	// just double the gas price for now, this is higher than needed for a replacement tx though
 	gasPrice := new(big.Int).Mul(tx.GasPrice(), big.NewInt(2))
-	cancelTx, err := inst.backend.ETHClient().CancelTxWithNonce(inst.backend.Ctx(), tx.Nonce(), gasPrice)
-	if err != nil {
-		return false, fmt.Errorf("failed to create or send cancel tx: %w", err)
-	}
-
-	log.Infof("submit cancel tx %s", cancelTx)
-
-	// TODO: if newSwap is included instead of the cancel tx (unlikely), block.WaitForReceipt will actually
-	// loop for 1 hour and block as there will never be a receipt for the cancel tx.
-	// we should probably poll for our account nonce to increase, and when it does, check for receipts
-	// on both txs to see which one was successful.
-	receipt, err := block.WaitForReceipt(inst.backend.Ctx(), inst.backend.ETHClient().Raw(), cancelTx)
+	receipt, err := inst.backend.ETHClient().CancelTxWithNonce(inst.backend.Ctx(), tx.Nonce(), gasPrice)
 	if err != nil && !errors.Is(err, ethereum.NotFound) {
 		return false, fmt.Errorf("failed to get cancel transaction receipt: %w", err)
 	}
 
-	if errors.Is(err, ethereum.NotFound) || receipt.Status == ethtypes.ReceiptStatusFailed {
```

Comment:
Our `block.WaitForReceipt` method doesn't return a receipt if the status failed. If you get a receipt, but the status is "failed", it means the transaction *was* included in a block despite the fact that it was reverted. Our `WaitForReceipt` method tries to pull the revert reason out of the block that the transaction was included in. I could be wrong, but I don't think we need to worry about it here. I don't think a simple transfer to a non-contract address can be reverted. If the gas limit is too low, the geth APIs won't even let you send the simple transfer into the mempool. The balance can't be too low, since we are sending 0 eth. The cancel tx should either get included successfully into a block, or not included at all. If I'm wrong, I'd be really curious to know how it can happen.